src/Maths/Silk.NET.Maths/Scalar.cs

```diff
@@ -0,0 +1,845 @@
+﻿// This file is part of Silk.NET.
+// 
+// You may modify and distribute Silk.NET under the terms
+// of the MIT license. See the LICENSE file for details.
+
+using System;
+using System.Runtime.CompilerServices;
+
+namespace Silk.NET.Maths
+{
+    internal static class Scalar<T> 
```

Comment:
Eventually I will want to look at making this a public API, but I'd need to put some time into thinking about what I want to include in it and formally spec up the API. It's good you've inadvertently made a start! So yeah, there isn't really a review comment here, other than at some point once we have an initial prototype I'd like to look at making this public. Future stuff though, all good for now.